99robots-header-footer-code-manager.php

```diff
@@ -1001,6 +1016,421 @@ public static function hfcm_request_handler()
+                );
+        
+                $offset = 0;
+                if(!empty($_POST['page'])) {
+                    $offset = 100 * sanitize_text_field($_POST['page']);
+                }
+                if(!empty($_POST['postType'])) {
+                    $postTypes = array(sanitize_text_field($_POST['postType']));
+                } else {
+                    $postTypes = array_diff(
+                        array_merge(array( 'post' ), get_post_types( $args, 'names', 'and' )), array('page')
+                    );
+                }
+        
+                if( isset( $_POST['page'] ) && 1 === $_POST['page'] ) {		
+                    $page = 1;
+                }
+                else {
+                    $page = $_POST['page'];
+                }
+                
+                $argsPost = array(
+                    'post_type' => $postTypes,
+                    'posts_per_page' => 5,
+                    'paged' => $page,
+                    'orderby' => 'title',
+                    // orderby created date
+                    'order' => 'DESC',
+                    //'order' => 'ASC',
+                    //'offset' => $offset,
+                    'post_status' => ['publish']
+                );
+        
+            
+                $searchQuery = "";
+                if(!empty($_POST['q'])) {
+                    $argsPost['s'] = sanitize_text_field($_POST['q']);
+                }
+        
+                if(!empty($_POST['taxonomy'])) {
+                    $taxonomySearch =  sanitize_text_field($_POST['taxonomy']);
+                    $taxonomySearch =  explode(':', $taxonomySearch);
+                    if(!empty($taxonomySearch)) {
+                        $argsPost['tax_query'] = array(
+                            array(
+                                'taxonomy' => $taxonomySearch[0],
+                                'field' => 'slug',
+                                'terms' => $taxonomySearch[1],
+                            )
+                        );
+                    }
+                }
+        
+        
+                $posts = get_posts($argsPost);
+        
+                $json_output = array(
+                    'selected' => array(),
+                    'posts' => '',
+                    'excluded' => array(),
+                );
+        
+                $selectOptions = "";
+                $selectizeResults = array();
+                foreach ( $posts as $pdata ) {
+                    $post_title = sanitize_text_field( $pdata->post_title );
+    
+                    if( $_POST['runFetchPosts'] ){
+        
+                        if( is_array( $ex_posts ) && !empty( $ex_posts ) && in_array( $pdata->ID, $ex_posts ) ){
+                            $selectOptions .= '<option class="left-side-option clone-button button-id-'.$pdata->ID.'" value="'.$pdata->ID.'|'.$pdata->post_title.'" disabled>'.sanitize_text_field( $pdata->post_title ).' + </option>';
```

Comment:
esc_attr? esc_html? are we not following the standard anymore? did you run this with plugin check for Wordpress? phpcs?

Reply:
do it now, and apply it for all